extensions/extensions-helm/aissemble-infrastructure-chart/templates/namespaces.yaml

```diff
@@ -0,0 +1,31 @@
+---
+{{- if index .Values "argo-cd" "enabled" }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ index .Values "argo-cd" "namespaceOverride" }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-10"
```

Comment:
Q: what does the -10 mean?

Reply:
The weight determines the order the kubernetes resources are installed in a helm chart. Each resource has a default weight of 0. The resources are then installed with the lowest weight first (negative to positive) So the weight of -10 is to ensure this resource is installed first, since it's the namespace that needs to exist before the other resources can be installed.